specs/ProgrammaticSaveAs.md

```diff
@@ -0,0 +1,429 @@
+                            showCustomizedDialog();
+                            CHECK_FAILURE(deferral->Complete());
+                        });
+                    return S_OK;
+                })
+                .Get(),
+            &m_saveAsRequestedToken);
+    }
+    m_hasSaveAsRequestedEventHandler = !m_hasSaveAsRequestedEventHandler;
+    MessageBox(
+        m_appWindow->GetMainWindow(),
+        (m_hasSaveAsRequestedEventHandler ? L"Event Handler Added" : L"Event Handler Rremoved"), L"Info",
+        MB_OK);
+    return true;
+}
+```
+### Programmatic Save As
+Call RequestSaveAs method to trigger the programmatic save as.
+```c++
+
+bool ScenarioSaveAs::ProgrammaticSaveAs()
+{
+    if (!m_webView2_20)
+        return false;
+    m_webView2_20->RequestSaveAs(
+        Callback<ICoreWebView2RequestSaveAsCompletedHandler>(
+            [this](HRESULT errorCode, COREWEBVIEW2_SAVE_AS_REQUESTED_RESULTS result) -> HRESULT
+            {
+                // Show RequestSaveAs returned result, optional
+                MessageBox(
+                    m_appWindow->GetMainWindow(),
+                    (L"Save As " + saveAsResultString[result]).c_str(), L"Info", MB_OK);
+                return S_OK;
+            })
+            .Get());
+    return true;
+}
+```
+
+## .Net/ WinRT
+### Add or Remove the Event Handler
+This example hides the default save as dialog and shows a customized dialog.
+```c#
+
+void TaggleEventHandlerExecuted(object target, ExecutedRoutedEventArgs e)
+{
+    if (hasSaveAsRequestedEventHandler)
+        webView.CoreWebView2.SaveAsRequested -= WebView_SaveAsRequested;
+    else 
+        webView.CoreWebView2.SaveAsRequested += WebView_SaveAsRequested;
+    hasSaveAsRequestedEventHandler = !hasSaveAsRequestedEventHandler;
+    MessageBox.Show(hasSaveAsRequestedEventHandler? "Event Handler Added":"Event Handler Rremoved" , "Info");
+}
+
+void WebView_SaveAsRequested(object sender, CoreWebView2SaveAsRequestedEventArgs args)
+    {
+    // Hide the system default save as dialog.
+    args.SuppressDefaultDialog = true;
+
+    // Developer can obtain a deferral for the event so that the CoreWebView2
+    // doesn't examine the properties we set on the event args until
+    // after the deferral completes asynchronously.
+    CoreWebView2Deferral deferral = args.GetDeferral();
+
+    // We avoid potential reentrancy from running a message loop in the event
+    // handler. Show the customized dialog later when complete the deferral
+    // asynchronously.
+    System.Threading.SynchronizationContext.Current.Post((_) =>
+    {
+        using (deferral)
+        {
+            // This is a customized dialog example, the constructor returns after the 
+            // dialog interaction is completed by the end user.
+            var dialog = new SaveAsDialog(_Kinds);
+            if (dialog.ShowDialog() == true)
+            {
+                // Preview the content mime type, optional.
+                args.ContentMimeType;
```

Comment:
No need to have this unused code getter that does nothing. But the comment can be added to the below comment and say something like 'You can use args.ContentMimeType to check the mime type of the document that will be saved to help setup your custom Save As dialog UI'

```suggestion
```